crates/key-value/src/lib.rs

```diff
@@ -193,7 +165,9 @@ impl spin_world::v1::key_value::Host for KeyValueDispatch {
     async fn get(&mut self, store: u32, key: String) -> Result<Result<Vec<u8>, LegacyError>> {
         let this = Resource::new_borrow(store);
         let result = <Self as key_value::HostStore>::get(self, this, key).await?;
-        Ok(result.map_err(to_legacy_error))
+        Ok(result
+            .map_err(to_legacy_error)
+            .map(|v| v.unwrap_or_default()))
```

Comment:
Should this map a `None` value to a `LegacyError::NoSuchKey`?

Reply:
Another good catch!